### PR TITLE
table format function

### LINKED ISSUE #564: Add formatting in tables using functions
Currently Taipy uses printf style formatting in tables
~~~ python
<|table|data={my_data}|columns={col_format}|...|>
col_format = {'Sales':'%.2f'}
~~~

This could be extended by allowing functions, like in <|selector|adapter={...}> 

~~~ python
col_format={ "Sales": {"format_function": format_func}}

def format_func(data, column_name) -> list[str]:
    ... apply formatting ...
    return formatted_column_data
~~~

Such functions might even return images 

Background:  
I'd like to have floats formatted like 9.999,99 which seems not possible using pure printf style syntax in Taipy.
Taipy doesn't support %'d to get locale aware thousands separator.
AFAIK printf doesn't support thousands separators in floats.

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #564

```
import pandas as pd

from taipy.gui import Gui

data = pd.DataFrame()

# Create 1O columns with 100 rows of bools
for i in range(10):
    data[f"col_{i}"] = list(range(100))


def my_format(state, value, index, row, col_name):
    return f"value: {value} for {index} in {col_name}"


page = """
<|{data}|table|format_fn[col_2]=my_format|not editable|>
"""
if __name__ == "__main__":
    Gui(page).run(title="564 table format function")

```

### CODE REVIEW

File: taipy/gui/utils/table_col_builder.py [hunk 1]
```diff
@@ -112,6 +112,21 @@ def _enhance_columns(  # noqa: C901
...
+    for k, v in formats.items():  # pragma: no cover
+        if col_desc := _get_column_desc(columns, k):
+            if callable(v):
+                value = hash_names.get(f"format_fn[{k}]")
+            elif isinstance(v, str):
+                value = v.strip()
+            else:
+                value = None
+            if value in columns.keys():
+                _warn(f"{elt_name}: format_fn[{k}]={value} cannot be a column's name.")
```
Comment (FabienLelaquais) [inline taipy/gui/utils/table_col_builder.py]: Not clear who 'cannot be a column's name'...

**Review by FabienLelaquais — DISMISSED** [review]
👍👍👍

File: taipy/gui/utils/table_col_builder.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply taipy/gui/utils/table_col_builder.py]: that's the value
I'll be happy to take a better wording

**Review by dinhlongviolin1 — DISMISSED** [review]
Nicely done!

File: taipy/gui/utils/table_col_builder.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply taipy/gui/utils/table_col_builder.py]: changed to `f"{elt_name}: format_fn[{k}] cannot reference a column's name '{value}'."`

**Review by FabienLelaquais — DISMISSED** [review]
👌

**Review by dinhlongviolin1 — DISMISSED** [review]

**Review by namnguyen20999 — DISMISSED** [review]
lgtm

**Review by FabienLelaquais — APPROVED** [review]
👍👍

**Comment by jrobinAV** [comment]
@FredLL-Avaiga This has been merged. Can we delete the branch?

**Pull request merged**